### ISSUE TITLE
What does EFV:NLI mean?

### ISSUE THREAD

**Issue by juppin** [MicrosoftDocs/OfficeDocs-o365seccomp#211]
My header is "X-Forefront-Antispam-Report: EFV:NLI;"

EFV is not listed in the list above...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 60b0c22e-ec20-d8dc-bbc5-cfbb9e98350e
* Version Independent ID: d9a1bac2-69ef-5fa7-23f1-417d7454c4cc
* Content: [Anti-spam message headers](https://docs.microsoft.com/en-us/office365/SecurityCompliance/anti-spam-message-headers)
* Content Source: [SecurityCompliance/anti-spam-message-headers.md](https://github.com/microsoftdocs/officedocs-o365seccomp/blob/public/SecurityCompliance/anti-spam-message-headers.md)
* Service: **o365-seccomp**
* GitHub Login: @KCCross
* Microsoft Alias: **krowley**

**Comment by AndreaBarr** [comment]
@juppin Thank you for this inquiry

I am not sure what this means either. Let me get this over to the o365seccomp team for investigation. 

Thank you for reporting and making the docs better. Much appreciated. 
 
I made a note to request the team to update this when the work is complete.

**Comment by Andy9999** [comment]
Hi, I got the same problem. What is funny is (sarcasm), that it started within an email communication with the same person that went back and forth a few times. The customer missed my reply because of that. It seems to affect 365 and hotmail as well. Is there any update on this?

**Comment by Andy9999** [comment]
If it helps in any way we joined the SNDS program and our IP addresses have normal status there. There is nothing else reported except EFV:NLI in the anti-spam header field.

**Comment by Andy9999** [comment]
We also have a valid SPF record in place.

**Comment by MSFTTracyP** [comment]
Thank you for your feedback on this. I've been talking about this issue internally. The article will be undergoing some updates.

**Comment by onspring-technologies** [comment]
How is this closed? THERE IS NO ANSWER?!?!

**Comment by Absoblogginlutely** [comment]
We're still getting efv:nli in headers too. I agree the document needs an overhaul, but this issue should not be closed until it is.

**Comment by rbianic** [comment]
Same message here, no documentation, spf/dkim/dmarc/dnssec/dane and my email view as spam by outlook...

**Comment by RichardKBR** [comment]
Same problem here X-Forefront-Antispam-Report: EFV:NLI;
My status is ok, in SNDS and other stuffs.

**Comment by denisebmsft** [comment]
Hello @onspring-technologies, @Absoblogginlutely, @rbianic, @RichardKBR, and everyone!

First of all, please accept our apologies for not being more clear in our closing this issue. The article has been updated a few times since this issue was first open, but we didn't explicitly address the EFV question in our responses here. We are sorry about that.

Second, we have looked into this further, and there is no current impact of the EFV:NLI header/verdict. It will be removed in the coming weeks.

We hope this helps clarify things, and once again, we apologize for any confusion.

**Comment by ktecho** [comment]
@denisebmsft I have that EFV:NLI in Outlook.com too, and my message went to Spam folder. If EFV:NLI don't have impact, how do I know why the email went to Spam?

**Comment by justinzhong** [comment]
As reported by others in this thread, I am receiving the following header when sending to outlook.live.com (hotmail address) and it is arriving in the Junk Email folder.

```
X-Forefront-Antispam-Report: EFV:NLI;
```
DKIM/SPF all passed, reverse IP record added for my mail server as well.

The same email send to Office 365 business accounts and Gmail accounts are delivered to inbox.

Would be appreciated if @denisebmsft from MSFT could please advise if EFV:NLI is not an issue, what other information we can use to troubleshoot further.

Thank you!

**Comment by onspring-technologies** [comment]
It might actually be more relevant if there were any documentation on the numeric spam codes. Literally the only report code we get is EFV:NLI, yet we're ending up in spam (when we're definitely not spam).  Yet we've got 5 numeric codes in the other header.

**Comment by amcorp1** [comment]
We are running into the same problem. Our emails are marked with EFV:NLI, and they're all ending up in the spam folder! Same emails get processed without any issues on other providers (Gmail, etc.).

**Comment by maximilianh** [comment]
Same here. BCL SCL are all 0, SPF is working. On stackoverflow someone suggests to send email only over IPv4 and use DKIM.

**Comment by maximilianh** [comment]
Contacting Microsoft worked but only for a week. We're sending email through the University server, Gmail servers, so definitely a well-known server with an address that has never been used for spam. The emails are signup confirmation emails, so if they land in spam, people cannot login to our website.

**Comment by kasinjsh** [comment]
I have this issue too. Only other mails service that lands My mail in Spam is Yahoo. I have SPF, DKIM, DMARC, PTR setup correctly. My servers IP address is not on any blacklist, I have checked a lot of them.

Mail still is marked whit EFV:NLI.

**Comment by ERKSagency** [comment]
The same thing is happening to me:
[https://www.mail-tester.com/test-02af9](https://www.mail-tester.com/test-02af9)

From Amazon SES (console -> TEST EMAIL) arrives at INBOX but from my script NO (only in Hotmail).

The strange thing is that by analyzing the headings, they are practically identical.

**Comment by kasinjsh** [comment]
> 
> 
> The same thing is happening to me:
> https://www.mail-tester.com/test-02af9
> 
> From Amazon SES (console -> TEST EMAIL) arrives at INBOX but from my script NO (only in Hotmail).
> 
> The strange thing is that by analyzing the headings, they are practically identical.

You don't have MX record record setup correctly. Fix that first.

**Comment by sponte** [comment]
I too seem to have this problem. For some time, Outlook.live.com seems to be very spam hungry and a lot of genuine messages end up in my junk folder. New services I signed up to recently, official BMW correspondence, virgin media. Looking into source of these messages, they all seem to have `EFV:NLI` in them.

Seems like the issue is more generic than just individual domains.

@AndreaBarr – has that been addressed properly?

**Comment by ERKSagency** [comment]
> > Me está pasando lo mismo: 
> > https://www.mail-tester.com/test-02af9
> > Desde Amazon SES (consola -> TEST EMAIL) llega a INBOX pero desde mi script NO (solo en Hotmail).
> > Lo extraño es que al analizar los encabezados, son prácticamente idénticos.
> 
> No tienes la configuración de registro de registro MX correctamente. Arregla eso primero.

MX records are already there, could you check it out? ... I have tried everything and still do not get to INBOX.

If you need an Amazon capture, mail header, or any configuration, let me know.

Thanks

**Comment by Andy9999** [comment]
Bump! This issue is still ongoing.

From a mail header of an email categorized as spam yesterday with EFV:NLI:
...
X-Forefront-Antispam-Report: EFV:NLI; 
...

SPF is being validated and is OK. No DKIM here. This should not go to spam.

**Comment by romain10009** [comment]
Hi everyone, I'm experiencing same issues, I tried sending the mail from gmail, from my own SMTP, and from mailgun. It keeps landing in spam.

Regards,

**Comment by ProBackup-nl** [comment]
Over here there is also an issue of `X-Forefront-Antispam-Report: EFV:NLI;` (and all messages going to spam).

Other headers (munged):
`Authentication-Results: spf=pass (sender IP is 8.1.5.1) smtp.mailfrom=s.nl; outlook.com; dkim=pass (signature was verified) header.d=s.nl;outlook.com; dmarc=pass action=none ...
`
helo does **not** match with rDNS of sender IP due to outbound link load balancing (outbound traffic is sent over multiple links of a Peplink Balance), helo=x.y.s.nl where rDNS of IPv4 = z.s.nl

`Received: from x.y.s.nl (8.1.5.1) by HE1EUR01FT003.mail.protection.outlook.com (10.152.0.89) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.1965.12 via Frontend Transport; Sun, 9 Jun 2019 09:09:02 +0000 `

Note: the TLS certificate has expired.

**Comment by Andy9999** [comment]
Hey ProBackup! Some good info. Unfortunately it did not solve our problem. We had an expired certificate from our own CA. I replaced it with a valid certificate. HELO name is OK and used in certificate and for rDNS. The only missing thing here is DKIM. So one can conclude from this that neither rDNS (ours is OK) nor DKIM (yours is present) solves the issue. Microsoft is so bad at email, I wonder why companies use their spamfilter in 365. They will have to get used to pick up the phone more frequently to ask email recipients if they really got the email. LOL. Here is a good tool to verify your TLS. Ours is all green/100%: www.checktls.com

**Comment by Andy9999** [comment]
In the header, we now, for whatever reason have a BCL of 5 meaning that there is some probability that recipients complain. WTF is this? I checked in Microsoft's SDNS and there the IP has normal status.

When will you hire some experts at mail Microsoft????

**Comment by Andy9999** [comment]
Another funny thing is, I remember that we at some time setup an Outlook365 test. For some reason MS hosts my email address on 365. I can logon to it and I even had 2 spam mails in my junk folder. The fun part is that we are self hosted. I sent an email to my hotmail test mailbox and it was received successfully. I opened two tickets, one for the BCL5 thing and another one for "stealing" my mail address.

**Comment by Andy9999** [comment]
Here is another unresolved case with the same set of rules in x-Microsoft-Antispam but BCL0:
https://answers.microsoft.com/en-us/outlook_com/forum/all/e-mail-in-spam-folder-without-reasoning/76fd1112-a449-4720-bdcf-d6f20ede8fba

**Comment by ProBackup-nl** [comment]
In our case it seems only the IP address that MS doesn't like.

**Comment by Andy9999** [comment]
Our IP is OK, at least in JMRP/SNDS.

@MS: Just in case if it helps to track the problem with the two junk mails I received on your servers using my self hosted email address.
1. Received: from DBXPR05CA0024.eurprd05.prod.outlook.com (2a01:111:e400:9434::24) by AM2PR05MB0898.eurprd05.prod.outlook.com (2a01:111:e400:842c::27) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256) id 15.1.991.14 via Mailbox Transport; Mon, 27 Mar 2017 22:12:28 +0000 Received: from inbound.mail.protection.outlook.com (213.199.154.182) by DBXPR05CA0024.outlook.office365.com (10.255.178.24) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256) id 15.1.991.14 via Frontend Transport; Mon, 27 Mar 2017 22:12:28 +0000 Received: from DB5EUR01FT055.eop-EUR01.prod.protection.outlook.com (10.152.4.56) by DB5EUR01HT082.eop-EUR01.prod.protection.outlook.com (10.152.5.181) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.977.7; Mon, 27 Mar 2017 22:12:26 +0000 Authentication-Results: spf=softfail (sender IP is 138.118.163.148) smtp.mailfrom=lindbergmanagement.com; sys-manage.com; dkim=none (message not signed) header.d=none;sys-manage.com; dmarc=none action=none header.from=lindbergmanagement.com; Received-SPF: SoftFail (protection.outlook.com: domain of transitioning lindbergmanagement.com discourages use of 138.118.163.148 as permitted sender) Received: from SNT004-MC3F23.hotmail.com (10.152.4.60) by DB5EUR01FT055.mail.protection.outlook.com (10.152.5.250) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.977.7 via Frontend Transport; Mon, 27 Mar 2017 22:12:25 +0000 X-IncomingTopHeaderMarker: OriginalChecksum:12F508FF1C512F7E1EF15A80E00C4C4A58EBADE1CDAE2449A815CA53091CA082;UpperCasedChecksum:EB49746AD1A0BC1FF81A766E2F6090AD9C47E45FB1E3E0A3ABFA6D4AC938984E;SizeAsReceived:1076;Count:17 Received: from lindbergmanagement.com ([138.118.163.148]) by SNT004-MC3F23.hotmail.com with Microsoft SMTPSVC(7.5.7601.23143); Mon, 27 Mar 2017 15:12:23 -0700 Received: from 185.127.25.138 with ESMTP; Tue, 28 Mar 2017 00:12:21 -0-100 

2.Received: from VI1PR0501CA0044.eurprd05.prod.outlook.com (10.168.132.30) by AM2PR05MB0898.eurprd05.prod.outlook.com (10.161.127.155) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.977.11 via Mailbox Transport; Sun, 26 Mar 2017 18:00:13 +0000 Received: from inbound.mail.protection.outlook.com (213.199.154.116) by VI1PR0501CA0044.outlook.office365.com (10.168.132.30) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256) id 15.1.991.14 via Frontend Transport; Sun, 26 Mar 2017 18:00:13 +0000 Received: from AM5EUR03FT031.eop-EUR03.prod.protection.outlook.com (10.152.16.57) by AM5EUR03HT165.eop-EUR03.prod.protection.outlook.com (10.152.17.244) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.977.7; Sun, 26 Mar 2017 16:50:48 +0000 Authentication-Results: spf=none (sender IP is 85.32.22.51) smtp.mailfrom=kaminfeger-erlenbach.ch; sys-manage.com; dkim=none (message not signed) header.d=none;sys-manage.com; dmarc=none action=none header.from=kaminfeger-erlenbach.ch; Received-SPF: None (protection.outlook.com: kaminfeger-erlenbach.ch does not designate permitted sender hosts) Received: from BAY004-MC5F13.hotmail.com (10.152.16.53) by AM5EUR03FT031.mail.protection.outlook.com (10.152.16.111) with Microsoft SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.977.7 via Frontend Transport; Sun, 26 Mar 2017 16:50:46 +0000 X-IncomingTopHeaderMarker: OriginalChecksum:A0193F9A312B4582A48D8FDE4C35ABACE80AF454A1C0EB08DB59A32910F08804;UpperCasedChecksum:FC7BEF0957D38C505B0D3987AB80B041CFFB844775E363CBE5C8014451C632F4;SizeAsReceived:880;Count:17 Received: from kaminfeger-erlenbach.ch ([85.32.22.51]) by BAY004-MC5F13.hotmail.com with Microsoft SMTPSVC(7.5.7601.23143); Sun, 26 Mar 2017 09:50:44 -0700 Received: from 185.23.17.10 with ESMTP; Sun, 26 Mar 2017 18:50:39 -0-100 

Both were in German language so it seems someone was sold a list of german outlook/hotmail/live/365 users. May it help you to track it down. Those two are the only mails in my junk folder and they are from 2017.

**Comment by Andy9999** [comment]
Just tried to delist the IP on https://sender.office.com. And this is the reply:
The IP address in question is not currently blocked in our system. Please refer to the email message you received from Microsoft and follow the steps it suggests.

**Comment by ProBackup-nl** [comment]
When inserting a smart host in the outbound mail server connection, the message from the same server with the same content is immediately delivered to the inbox.

I think the indicator is: `RF:JunkEmail; ` with nothing appended.

**Comment by Andy9999** [comment]
OK, thanks. So a copy of an outbound mail is delivered to a mailbox on the sending server and then tagged as spam?

**Comment by Andy9999** [comment]
Our BCL problem has been solved and it is zero again after contacting MS here:
https://support.microsoft.com/en-us/supportrequestform/8ad563e3-288e-2a61-8122-3ba03d6b8d75

And there, I think this one is only needed if the IP is black listed which wasn't the case for us:
https://sender.office.com/

Also it seems mail gets delivered again. I hope it stays like this.

I also removed my posting about the two junk mails in the mailbox using my alias name, hosted on the MS systems, even though our MX records never pointed there. Somehow the mailbox got created when signing up for the Windows/Windows Phone store, where an AD tenant was created. It seems the mails got there by a "stupid" MTA written by a spammer, trying each recipient with each mail server of the recipient list, therefore the MS systems picked up the mail, because they found such an alias. So this was kind of a false alert.

**Comment by Andy9999** [comment]
Here is another cool tool to test your emails:
https://www.mail-tester.com

We were not yet using DMARC and I now added an entry to our DNS hoping it increases the probability of successfully delivering mail to MS mail services, having a lower probability than any other mail provider (aka under performer).

**Comment by Andy9999** [comment]
Scam:

Hello,

My name is XXXXXXX and I work with the Outlook.com Deliverability Support Team.

I do not see anything offhand with the IPs: (1.2.3.4) that would be preventing your mail from reaching our customers. 

You may be able to find additional information on common delivery questions at the Outlook.com Postmaster Site found at http://postmaster.msn.com/ I would like to highlight some key areas that I believe are appropriate to your company.

·         Outlook.com has created the Smart Network Data Services (SNDS) program. This is a service that helps legitimate email senders work with their customers and partners to reduce spam originating from their IP. To register, please go to http://postmaster.msn.com/snds <http://postmaster.msn.com/snds> / This program allows a sender to monitor the 'health' of their IPs.    

·         Monitor user complaints. Outlook.com also has a sender complaint feedback loop program called the Junk Email Reporting Program (JMRP). Enrollment in this free program will benefit you as a sender as it will keep your email lists updated and populated with interested Outlook.com customers. This program will help you to remove those Outlook.com customers who do not want to receive emails from your company. If you are interested in joining this program, please visit https://postmaster.live.com/snds/JMRP.aspx?wa=wsignin1.0 <https://postmaster.live.com/snds/JMRP.aspx?wa=wsignin1.0> .

While using the SNDS tool and enrollment in the JMRP will not allow emails from your mail servers to bypass our filters, these are in place to help legitimate companies deliver their emails to Outlook.com customers.

**Apply for the Sender Score Certified Mail program. 
If you are doing all the above and you continue to have deliverability issues, you may wish to consider joining the Sender Score Certified Mail Program, a third party program administered by Return Path, Inc. Many legitimate mailers and marketers have qualified and joined this program to improve mail deliverability and decrease email from being filtered to the Junk E-mail Folder. Sender Score (www.senderscorecertified.com <http://www.senderscorecertified.com/> ) is the only service to which we subscribe**

 

The troubleshooting steps in this email are recommendations only. Microsoft makes no guarantees that following these steps will guarantee deliverability to MSN, Outlook.com, or Live.com customers.  

For more detailed information about best sending practices to Outlook.com users, please review the following white paper: http://download.microsoft.com/download/e/3/3/e3397e7c-17a6-497d-9693-78f80be272fb/enhance_deliver.pdf <http://download.microsoft.com/download/e/3/3/e3397e7c-17a6-497d-9693-78f80be272fb/enhance_deliver.pdf>   

Sincerely,

XXXXXXXXXX
Outlook.com Deliverability Support Team.

**Comment by rocketraman** [comment]
I have another example of an email which, by all the other headers, should have ended up in the user's inbox, but did not, and has the mysterious `EFV:NLI` value in the `X-Forefront-Antispam-Report` header.

Despite using DKIM, SPF, and DMARC (all pass), seeing `SFV:NSPM` (not spam), `SCL:1` (spam score level 1) in the `X-Forefront-Antispam-Report` header, and `BCL:0`, and `PCL:0` in the `X-Microsoft-Antispam` header, and absolutely no other indication of a problem in the received email, somehow this email still ended up in the user's spam folder.

Searching for this, I see lots of users with the same issue, but always the same useless non-answers.

**Comment by Andy9999** [comment]
Funny, the next mail had the return path link updated with a working one.

Here is what some people have to say about sender score / return path: https://www.complaintsboard.com/complaints/return-path-c276871.html#comments

Nothing good.

**Comment by Andy9999** [comment]
Scam/highway robbery for like the past 10 years or more with hotmail and possibly with 365 now too.

The link mentioned "http://www.senderscorecertified.com/" is 404ing by the way, maybe you want to update it:

Hello,

My name is XXXXXXX and I work with the Outlook.com Deliverability Support Team.

I do not see anything offhand with the IPs: (1.2.3.4) that would be preventing your mail from reaching our customers.

You may be able to find additional information on common delivery questions at the Outlook.com Postmaster Site found at http://postmaster.msn.com/ I would like to highlight some key areas that I believe are appropriate to your company.

· Outlook.com has created the Smart Network Data Services (SNDS) program. This is a service that helps legitimate email senders work with their customers and partners to reduce spam originating from their IP. To register, please go to http://postmaster.msn.com/snds http://postmaster.msn.com/snds / This program allows a sender to monitor the 'health' of their IPs.

· Monitor user complaints. Outlook.com also has a sender complaint feedback loop program called the Junk Email Reporting Program (JMRP). Enrollment in this free program will benefit you as a sender as it will keep your email lists updated and populated with interested Outlook.com customers. This program will help you to remove those Outlook.com customers who do not want to receive emails from your company. If you are interested in joining this program, please visit https://postmaster.live.com/snds/JMRP.aspx?wa=wsignin1.0 https://postmaster.live.com/snds/JMRP.aspx?wa=wsignin1.0 .

While using the SNDS tool and enrollment in the JMRP will not allow emails from your mail servers to bypass our filters, these are in place to help legitimate companies deliver their emails to Outlook.com customers.

Apply for the Sender Score Certified Mail program.
If you are doing all the above and you continue to have deliverability issues, you may wish to consider joining the Sender Score Certified Mail Program, a third party program administered by Return Path, Inc. Many legitimate mailers and marketers have qualified and joined this program to improve mail deliverability and decrease email from being filtered to the Junk E-mail Folder. Sender Score (www.senderscorecertified.com http://www.senderscorecertified.com/ ) is the only service to which we subscribe

The troubleshooting steps in this email are recommendations only. Microsoft makes no guarantees that following these steps will guarantee deliverability to MSN, Outlook.com, or Live.com customers.

For more detailed information about best sending practices to Outlook.com users, please review the following white paper: http://download.microsoft.com/download/e/3/3/e3397e7c-17a6-497d-9693-78f80be272fb/enhance_deliver.pdf http://download.microsoft.com/download/e/3/3/e3397e7c-17a6-497d-9693-78f80be272fb/enhance_deliver.pdf

Sincerely,

XXXXXXXXXX
Outlook.com Deliverability Support Team.

**Comment by Andy9999** [comment]
Funny, the next mail had the return path link updated with a working one. They don't even have prices on their web site and it looks to me like a tool for email marketers.

Here is what some people have to say about sender score / return path: https://www.complaintsboard.com/complaints/return-path-c276871.html#comments

Doesn't look good.

Anyways if checking our domain for black lists (including sender score) it's all good. So that is not the problem.

**Comment by Andy9999** [comment]
Our mails are not going to Junk Mail anymore.

We did setup DMARC, which is most likely not necessary and DKIM.

Before we had a 9/10, because of missing DKIM on https://www.mail-tester.com and now it is 10/10. If you don't send out bulk-mails, like us, you can safely ignore the "Your message does not contain a List-Unsubscribe header" warning. Anything else is green in our case.

Also, if you use TLS, you may want to check https://www.checktls.com/, because we found that we had an expired certificate. We score 100% for all tests there too.

The mails now pass, the mail headers do not contain RF:JunkEmail anymore in the "X-Microsoft-Antispam-Mailbox-Delivery:" header.

The mail still has EV:NLI, but it does not affect delivery.

I hope this helps!

**Comment by syarkhan** [comment]
The EFV:NLI; probably has nothing to do with your email landing in junk.
I had the same problem as my legit emails were landing in junk folder and then I removed these headers:
1. X-Fbl
2. 'List-Unsubscribe-Post': 'List-Unsubscribe=One-Click' 
3. X-Complaints-To
4. List-Id
5. X-Mailer

After removing these headers, my emails are now landing in Inbox.

I also did some tests with GlockApps and before removing these headers, all the other providers(gmail, yahoo) were marking my email as inbox but hotmail and outlook were marking it as Spam straightaway.

After removing these headers, outlook and hotmail marked it as Inbox.

Hope this helps people whose legit emails are landing in spam.
